sdk/tables/Azure.Data.Tables/src/TableClient.cs

```diff
@@ -377,7 +382,12 @@ public virtual TableSasBuilder GetSasBuilder(TableSasPermissions permissions, Da
         /// <returns>An instance of <see cref="TableSasBuilder"/>.</returns>
         public virtual TableSasBuilder GetSasBuilder(string rawPermissions, DateTimeOffset expiresOn)
         {
-            return new(Name, rawPermissions, expiresOn) { Version = _version };
+            TableSasBuilder builder = new(Name, rawPermissions, expiresOn);
+            if (!_isCosmosEndpoint)
```

Comment:
Why ignore for cosmos?

Reply:
Cosmos doesn't support the latest version for SAS auth.